src/studio/src/designer/backend/Services/Implementation/RepositorySI.cs

```diff
@@ -1100,6 +1100,29 @@ public async Task<RepositoryClient.Model.Repository> CreateService(string org, S
             return repository;
         }
 
+        /// <summary>
+        /// Deletes the local repository for the user and makes a new clone of the repo
+        /// </summary>
+        /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
+        /// <param name="repositoryName">the name of the local repository to reset</param>
+        /// <returns>True if the reset was successful, otherwise false.</returns>
+        public bool ResetLocalRepository(string org, string repositoryName)
+        {
+            string userName = AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext);
+            string repoPath = _settings.GetServicePath(org, repositoryName, userName);
+
+            if (Directory.Exists(repoPath))
+            {
+                // "Soft-delete" of local repo folder with same name to make room for clone of the new repo
+                string backupPath = _settings.GetServicePath(org, $"{repositoryName}_REPLACED_BY_NEW_CLONE_{DateTime.Now.Ticks}", userName);
```

Comment:
DeleteService in the same SI is currently not in use but might be at a later point.. should we add a comment about deleting softDeleted previous versions of clones as well?